src/runtime/tests/field-path-test.ts

```diff
@@ -356,15 +357,15 @@ describe('field path validation', () => {
       resolveFieldPathType(['inlined', 'name'], type);
     });
 
-    it('rejects missing fields nested inside inline schemas', async () => {
+    it('rejects missing fields nested inside inline schemas', Flags.withFlags({recursiveSchemasAllowed: true}, async () => {
```

Comment:
It'd be good to have a few tests to verify that it throws an exception when the flag is disabled

Reply:
Added a few more tests, I think I've covered most cases, should revisit it when new types are added like unions or nullables.